Azure/azure-cli
# [DO NOT MERGE] [SQL] Add managed instance DNS alias commands

**Related command**
az sql mi dns-alias
This is a new command

**Description**
This PR adds 5 new **atomic** commands for managing Managed Instance DNS aliases:
new, update, show, list, acquire
All commands support --ids parameters - check the examples in the command files.
Please **review** only, this should not be merged yet, but it should be ready for merge in August.

**Testing Guide**
az sql mi dns-alias new -g <resourceGroupName> --mi <managedInstanceName> --name <dnsAliasName>
az sql mi dns-alias show -g <resourceGroupName> --mi <managedInstanceName> --name <dnsAliasName>
az sql mi dns-alias update -g <resourceGroupName> --mi <managedInstanceName> --name <dnsAliasName> --has-dns-alias
az sql mi dns-alias list -g <resourceGroupName> --mi <managedInstanceName>
az sql mi dns-alias acquire -g <targetResourceGroupName> --mi <targetManagedInstanceName> --source-id <sourceDnsAliasResourceId>

Tests are also added, so they can be checked out.

This checklist is used to make sure that common guidelines for a pull request are followed.
